src/server/editorServices.ts

```diff
@@ -192,6 +192,10 @@ namespace ts.server {
             return this.roots.map(root => root.fileName);
         }
 
+        getScriptKind() {

```

Comment:
getScriptKind() is a way to retrieve the script kind from the host when it cannot be determined by the file name.  For example, the VS host supports script block files which can have the following extensions (.html, .htm, .aspx, .ascx, etc).The VS host returns ScriptKind.JS for these file extensions. For other hosts that do not currently require support for script block files ScriptKind.Unknown is returned. For these hosts if the script kind cannot be determined from the file name and ScriptKind.Unknown is encountered, the default ScriptKind.TS will be used.